### ISSUE TITLE
Failure if an existing lvm is found

### ISSUE THREAD

**Issue by BigAl** [picodotdev/alis#132]
This is slightly tricky as I'm always ware of messing with disks. However if I do a reinstall on a previous system that already  had an lvm  setup the `umount` command on this line fails
https://github.com/picodotdev/alis/blob/6966bb3c7052c6dd69f5fa753cdcbe2e62c9c28c/alis.sh#L349

This would be expected as the filesystem should not and is not mounted. Howvever the boot kernel on the install disk has found and dejected and existing Volume Group

If think what needs/should be done here is to remove the logical volume and and remove the volume group

```sh
        lvremove "/dev/$LVM_VOLUME_GROUP/$LVM_VOLUME_LOGICAL"
        vgremove "/dev/$LVM_VOLUME_GROUP"
```

No need to umount the disk.

**Comment by BigAl** [comment]
See PR https://github.com/picodotdev/alis/pull/133 fro possible solution

**Comment by picodotdev** [comment]
The objetive with this function was to allow a reinstall on same disk, when alis fails and is rexecuted (partitions can be mounted) or booting from a power off system (partitions will not be mounted).

I changed the prepare_partition so umounts dont fail checking mounts properly with mountpoint, if this is enough for you?.

https://github.com/picodotdev/alis/commit/d40ce427c9d55747ec00c2330ba49f65d19abc13

With the PARTITION_MODE set to auto if the LVM volumes (logical, group and phisical) exist should not be a problem as the disk will be wiped https://github.com/picodotdev/alis/blob/ae3876df828e26e414ab1356851a2c0f8dcde59e/alis.sh#L539 (but I dont know if this works completely as expected).

**Comment by BigAl** [comment]
Looked at the LOG and yes you still need to do a `lvremove` and `vgremove` I'm pretty sure this is because they are still in the boot kernel memory probably.

I think adding the `lvremove` and vgremove` somewhere in that block with `sgdisk --zap-all $DEVICE` will do the trick.

**Comment by picodotdev** [comment]
You were right.

I did testing using LVM on LUKS not only LVM, in the former case the behavior was different, no necesary to remove LVM volmes, with only deactivate and close LUKS seems was enough. 

Now I hope this edge case but possible is working as expected.

I did the (lv|vg|pv)remove before creating VLM volumes, I want that data destructive operations to be only in partition function not also in prepare_partition.
